### PR TITLE
Typescript definition

### PR DESCRIPTION
**Pull request by jucrouzet**
Adds typescript definition to use module in .ts files natively.

### CODE REVIEW

**Comment by jucrouzet** [comment]
Hi @indexzero (or any maintainer),
This PR only adds a definition for TS, it does not expose any new features or change an existing one, the goal is only to avoid an external dep in DefinitivelyTyped, is this possible for you to merge it quick ?
If you don't want to handle TS, let me know, I'll create a DefinitivelyTyped def, no problem.

Thanks !

File: lib/portfinder.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * The lowest port to begin any port search from.
+ */
+export declare let basePort: number;
+
+/**
+ * Responds with a unbound port on the current machine.
+ */
+export declare function getPort(callback: PortfinderHandler): void;
```
Comment (eriktrom) [inline lib/portfinder.d.ts]: @jucrouzet - thanks a bunch - `getPort` actually takes an optional options object and a callback. The options are `host: string` and `port: number` - do u mind adding those? (https://github.com/indexzero/node-portfinder/blob/c5dd2cc76bd26d4771cee9e0637dfa27a7f89f24/lib/portfinder.js#L25-L28)

also - if u pass a port option, it takes precedence over the `exports.basePort` setting - https://github.com/indexzero/node-portfinder/blob/c5dd2cc76bd26d4771cee9e0637dfa27a7f89f24/lib/portfinder.js#L30
(not sure how u declare that in TS, if not possible, no worries)

mostly, do u mind adding the options object?

then i'll merge this in, thanks again

Comment (jucrouzet) [inline lib/portfinder.d.ts]: Ok, thought that it was not "public" as it was not in the README.
The fact that the `port` option takes precedence over `basePort` has no implication in definition but I add it on comment for reference.

Comment (eriktrom) [inline lib/portfinder.d.ts]: awesome thank you

yeah the port and host are public, README needs a lil update

(technically u can pass a server option too, that is NOT public :) )

again thanks for the commit

Comment (jucrouzet) [inline lib/portfinder.d.ts]: YW
Embed TS definitions are a lot safer than DefinitivelyTyped ones ...

**Comment by jucrouzet** [comment]
Sorry @eriktrom I went a little too fast on this one, got to make options optional, can you reopen the PR ?

**Comment by eriktrom** [comment]
@jucrouzet - no problem - b/c it was just type declarations i pushed to npm already, open another pr with the fixes - thanks for taking the time to add that - i'll merge and re-publish once ur ready

**Pull request merged**